lib/adf_info.py

```diff
@@ -207,136 +220,144 @@ def __init__(self, config_file, debug=False):
                     eyear_baseline = found_eyear_baseline
-            #End if
-
-            #Check if history file path exists:
-            if baseline_hist_locs:
-
-                starting_location = Path(baseline_hist_locs)
-                file_list = sorted(starting_location.glob('*'+hist_str+'.*.nc'))
-                #Partition string to find exactly where h-number is
-                #This cuts the string before and after the `{hist_str}.` sub-string
-                # so there will always be three parts:
-                # before sub-string, sub-string, and after sub-string
-                #Since the last part always includes the time range, grab that with last index (2)
-                #NOTE: this is based off the current CAM file name structure in the form:
-                #  $CASE.cam.h#.YYYY<other date info>.nc
-                base_climo_yrs = [int(str(i).partition(f"{hist_str}.")[2][0:4]) for i in file_list]
-                base_climo_yrs = sorted(np.unique(base_climo_yrs))
-
-                base_found_syr = int(base_climo_yrs[0])
-                base_found_eyr = int(base_climo_yrs[-1])
-
-                #Check if start or end year is missing. If so then just assume it is the
-                #start or end of the entire available model data.
-                if syear_baseline is None:
-                    msg = f"No given start year for {data_name}, "
-                    msg += f"using first found year: {base_found_syr}"
-                    print(msg)
-                    syear_baseline = base_found_syr
-                if syear_baseline not in base_climo_yrs:
-                    msg = f"Given start year '{syear_baseline}' is not in current dataset "
-                    msg += f"{data_name}, using first found year: {base_climo_yrs[0]}\n"
-                    print(msg)
-                    syear_baseline = base_found_syr
-
-                if eyear_baseline is None:
-                    msg = f"No given end year for {data_name}, "
-                    msg += f"using last found year: {base_found_eyr}"
-                    print(msg)
-                    eyear_baseline = base_found_eyr
-                if eyear_baseline not in base_climo_yrs:
-                    msg = f"Given end year '{eyear_baseline}' is not in current dataset "
-                    msg += f"{data_name}, using last found year: {base_climo_yrs[-1]}\n"
-                    print(msg)
-                    eyear_baseline = base_found_eyr
-
-                #Grab baseline nickname
-                base_nickname = self.get_baseline_info('case_nickname')
-                if base_nickname is None:
-                    base_nickname = data_name
-            #End if
-
-            #Grab baseline nickname
-            base_nickname = self.get_baseline_info('case_nickname')
+            # End if
+
+            # Check if history file path exists:
+            if any(baseline_hist_locs):
+                for hist_str in baseline_hist_str:
```

Comment:
You're right in that the values aren't retained. It turns out it doesn't need                                                               
to be a loop because we assume all the files in a case have the same dates,                                                         
and we are only using one syear/eyear per case, not per hist_str.                                                                   
So I've changed this to                                                                                                             
 ```
               hist_str = baseline_hist_str[0]                                                                                     
                hist_str = hist_str_case[0]                
```                                                                         
and removed the for-loop over hist_str.                                                                                             
(Will commit with the other changes)